src/Parameter.h

```diff
@@ -55,6 +60,15 @@ class Parameter {
      * explicitly specified (as opposed to autogenerated). */
     Parameter(const Type &t, bool is_buffer, int dimensions, const std::string &name);
 
+    /** Construct a new parameter via deserialization. */
+    Parameter(const Type &t, bool is_buffer, int dimensions, const std::string &name,
+              const Buffer<void> &buffer, int host_alignment, const std::vector<BufferConstraint> &buffer_constraints,
+              MemoryType memory_type);
+
+    Parameter(const Type &t, bool is_buffer, int dimensions, const std::string &name,
```

Comment:
Needs a doc string. Note that in doxygen output this won't necessarily appear in a way that makes it clear that the comment on the previous constructor also applies.